src/main/java/uk/gov/justice/hmpps/casenotes/filters/OffenderCaseNoteFilter.kt

```diff
@@ -0,0 +1,120 @@
+package uk.gov.justice.hmpps.casenotes.filters
+
+import com.google.common.collect.ImmutableList
+import jakarta.persistence.criteria.CriteriaBuilder
+import jakarta.persistence.criteria.CriteriaQuery
+import jakarta.persistence.criteria.Join
+import jakarta.persistence.criteria.JoinType
+import jakarta.persistence.criteria.Predicate
+import jakarta.persistence.criteria.Root
+import lombok.Builder
+import lombok.EqualsAndHashCode
+import org.apache.commons.lang3.StringUtils
+import org.springframework.data.jpa.domain.Specification
+import uk.gov.justice.hmpps.casenotes.model.OffenderCaseNote
+import java.time.LocalDateTime
+import java.util.function.Consumer
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
+@Builder
+@EqualsAndHashCode
+class OffenderCaseNoteFilter(
+  private val offenderIdentifier: String? = null,
+  private val locationId: String? = null,
+  private val authorUsername: String? = null,
+  private val excludeSensitive: Boolean = false,
+  private val startDate: LocalDateTime? = null,
+  private val endDate: LocalDateTime? = null,
+  private val typeSubTypes: List<String> = emptyList(),
+) : Specification<OffenderCaseNote> {
+
+  override fun toPredicate(root: Root<OffenderCaseNote>, query: CriteriaQuery<*>, cb: CriteriaBuilder): Predicate? {
+    val predicateBuilder: ImmutableList.Builder<Predicate> = ImmutableList.builder()
```

Comment:
I switched this in Prison API to use `mutableListOf` rather than using google classes